docs/mkdocs/lab.md

```diff
@@ -1,207 +1,376 @@
-Once the module is built, copy it in a next step and create the definitive image which will be uploaded to the OpenShift
-internal registry and used as the Module Loader in the selected nodes.
-This multi-stage example could be useful if you need the use of extra software which is not present at DTK image but it
-is in the destination image on which the module files are copied.
+When using DTK, we should set which version of the Driver Toolkit [image](https://github.com/openshift/driver-toolkit#finding-the-driver-toolkit-image-url-in-the-payload) we are using to match the exact kernel version running on OpenShift nodes. In our case, we are using DTK with KMM and KMM will automatically add the appropriate ${DTK_AUTO} value.
 
-All `Dockerfile` steps needed in order to build a module from sources should be configured in a `ConfigMap` defined
-previous to the Module object creation.
+We can get further info at DTK [repository](https://github.com/openshift/driver-toolkit#readme).
 
-<details>
-<summary>Full example</summary>
 
+## Building In-Cluster Modules from Sources
+
+In this example, we will demonstrate how to build the kernel module in our cluster using sources from a git repository. The process involves several steps:
+
+1) Adding a ConfigMap: We will create a ConfigMap to store all the necessary information for the build process, including the image we are using (DTK in this case), the remote git repository containing the sources, and the compiling commands.
+
+2) Creating the Module object: Next, we'll create the Module object itself. It will configure the registry for the resulting image, set up a secret for possible authentication, specify the previously created ConfigMap for use, and define the nodes where the pods should be scheduled.
+
+Applying next file in an OpenShift cluster should build and load a kmm-ci-a module in the nodes labeled with a `worker` role:
+
+**in_tree_build.yaml**
+
+```yaml
+    ---
+    apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: build-module-labs
+      namespace: openshift-kmm
+    data:
+      dockerfile: |
+        ARG DTK_AUTO
+        ARG KERNEL_VERSION
+        FROM ${DTK_AUTO} as builder
+        WORKDIR /build/
+        RUN git clone -b main --single-branch https://github.com/rh-ecosystem-edge/kernel-module-management.git
+        WORKDIR kernel-module-management/ci/kmm-kmod/
+        RUN make
+        FROM docker.io/redhat/ubi8-minimal
+        ARG KERNEL_VERSION
+        RUN microdnf -y install kmod openssl && \
+            microdnf clean all && \
+            rm -rf /var/cache/yum
+        RUN mkdir -p /opt/lib/modules/${KERNEL_VERSION}
+        COPY --from=builder /build/kernel-module-management/ci/kmm-kmod/*.ko /opt/lib/modules/${KERNEL_VERSION}/
+        RUN /usr/sbin/depmod -b /opt
+    ---
+    apiVersion: kmm.sigs.x-k8s.io/v1beta1
+    kind: Module
+    metadata:
+      name: kmm-ci-a
+      namespace: openshift-kmm
+    spec:
```

Comment:
Should we add the `version` field that was added in 1.1?

Reply:
I just saw that you have a separate example for ordered upgrades.
Please ignore my comment :)